tutorial/001-the-host/README.md

```diff
@@ -0,0 +1,79 @@
+# The libp2p 'host'
+For most applications, the host is the basic building block you'll need to get
+started. This guide will show how to construct and use a simple host.
+The host is an abstraction that manages services on top of a swarm. It provides
+a clean interface to connect to a service on a given remote peer.
+
+First, you'll need an ID, and a place to store that ID. To generate a 'test'
+ID, you can do the following:
+```go
+import (
+	testutil "github.com/libp2p/go-testutil"
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+)
+
+// For toy applications, this is an easy way to get an identity
+ident, err := testutil.RandIdentity()
+if err != nil {
+	panic(err)
+}
+
+// We've created the identity, now we need to store it.
+// A peerstore holds information about peers, including your own
+ps := pstore.NewPeerstore()
+
+// An identity is essentially a public/private keypair
+ps.AddPrivKey(ident.ID(), ident.PrivateKey())
+ps.AddPubKey(ident.ID(), ident.PublicKey())
+```
+
+Next, you'll need at least one address that you want to listen on. You can go
+from a string to a multiaddr like this:
+```go
+import ma "github.com/multiformats/go-multiaddr"
+
+...
+
+maddr, err := ma.NewMultiaddr("/ip4/0.0.0.0/tcp/9000")
```

Comment:
In this case I *do* want to bind to all interfaces. This is a swarm listener address